rfcs/0005-music-system.md

```diff
@@ -0,0 +1,155 @@
+5. Upon leaving the system, the `playlist` becomes invalid as any other would. At this time, the standard `playlists` would no longer be treated as invalid, and would be chosen to replace the system's music.
+
+When a mission trigger plays a `track`, the behaviour is different:
+1. The current `track` stops playback. The active `playlist` is discarded.
+2. The mission's `track` starts playing from the beginning. `Variant`-related playback rules are NOT followed.
+3. When the mission's `track` finishes, a new `playlist` is chosen as usual. There is no guarantee tha the previously playing `playlist` will be resumed here.
+
+> [!NOTE]
+> An experimental demo of this system is available at https://tibetiroka.com/music with limited functionality.
+
+# Drawbacks
+Having multiple music `layers` increases complexity and file I/O operations. It can also increase the number of audio output sinks used, though we can mix the music samples before sending them to the operating system if necessary.
+
+As music is often streamed, there is no easy way to guarantee that an audio file begins playback immediately. Therefore there is no way to keep the foreground layers in sync with the background layer completely without using cached WAV samples. (The foreground layers are meant to provide ambience, not orchestral playback, so this is probably a non-issue.)
+
+# Alternatives
+Alternatives have been discussed extensively regarding the previous music PRs. Though much of the structure is preserved from the previous solutions, there are some key differences:
+- `Variants` are now for `tracks`, not for audio files. They are also arbitrarily defined, and are not required for the combat/non-combat switching to work properly.
+  - As it isn't reasonable to expect every `track` to have a combat `variant`, the new approach enables playing dedicated combat music in scenarios where previously regular music would have been played. This is helped by moving the combat `variants` to another `playlist` from the file specification of `tracks`.
+- `Tracks` can have multiple audio layers playing concurrently.
+  - This choice was made based on the results of the creator survey, and was not known at the time of the previous discussions.
+- `Playlists` are now "play once", not "play forever". This fixes various issues where players could get stuck listening to the same few songs when many others were also available.
+  - This change makes it much easier to have, for example, "generic human" tracks in a dedicated `playlist` while still having dedicated `tracks` in each region's own `playlist`.
+
+# Unresolved Questions
+Should `variant` support more than two `tracks` for convenience? Is there a real need?
+
+Should `playlists` have playback modes other than "shuffle"? Though we have experienced extensive issues with ordered playback, some other playback modes still might be viable.
+
+Is there a need to add weights to `playlists`?
```

Comment:
At least there is no downside to deferring it, a lack of weights makes for a sensible default.